src/metabase/models/card.clj

```diff
@@ -126,15 +128,49 @@
   [_model k cards]
   (mi/instances-with-hydrated-data
    cards k
-   #(->> (t2/query {:select    [[:%count.* :count] :card_id]
-                    :from      [:report_dashboardcard]
-                    :where     [:in :card_id (map :id cards)]
-                    :group-by  [:card_id]})
-         (map (juxt :card_id :count))
-         (into {}))
+   (fn []
+     (->> (t2/query {:select    [[:%count.* :count] :card_id]
+                     :from      [:report_dashboardcard]
+                     :where     [:in :card_id (map :id cards)]
+                     :group-by  [:card_id]})
+          (map (juxt :card_id :count))
+          (into {})))
    :id
    {:default 0}))
 
+(methodical/defmethod t2/batched-hydrate [:model/Card :in_dashboards]
+  [_model k cards]
+  (mi/instances-with-hydrated-data
+   cards k
+   (fn []
+     (->> (t2/query {:select [:card_id
+                              :name
+                              :collection_id
+                              :id]
+                     :from [[{:union-all [{:select [[:dc.card_id :card_id]
+                                                    [:d.name :name]
+                                                    [:d.collection_id :collection_id]
+                                                    [:d.id :id]]
+                                           :from [[:report_dashboardcard :dc]]
+                                           :join [[:report_dashboard :d] [:= :dc.dashboard_id :d.id]]
+                                           :where [:in :dc.card_id (map :id cards)]}
+                                          {:select [[:dcs.card_id :card_id]
+                                                    [:d.name :name]
+                                                    [:d.collection_id :collection_id]
+                                                    [:d.id :id]]
+                                           :from [[:dashboardcard_series :dcs]]
+                                           :where [:in :dcs.card_id (map :id cards)]
+                                           :join [[:report_dashboardcard :dc] [:= :dc.id :dcs.dashboardcard_id]
+                                                  [:report_dashboard :d] [:= :d.id :dc.dashboard_id]]}]}
+                             :dummy_alias]]})
+          (group-by :card_id)
+          (m/map-vals (fn [dashes] (->> dashes
+                                        (map (fn [dash] (dissoc dash :card_id)))
+                                        distinct
```

Comment:
should this be `dedupe`?

Reply:
I don't think so? `(dedupe [:a :b :a])` returns `(:a :b :a)` (because it only removes consecutive duplicates), where `(distinct [:a :b :a])` returns `(:a :b)`.